source/dagger/gameplay/tank_warfare/collectibles.cpp

```diff
@@ -0,0 +1,134 @@
+#include "collectibles.h"
+
+#include "core/core.h"
+#include "core/engine.h"
+#include "core/input/inputs.h"
+#include "core/graphics/sprite.h"
+#include "core/graphics/animation.h"
+#include "core/graphics/shaders.h"
+#include "core/graphics/window.h"
+#include "core/game/transforms.h"
+
+#include "gameplay/common/simple_collisions.h"
+#include "gameplay/tank_warfare/tank_controller.h"
+
+using namespace tank_warfare;
+
+void CollectibleSystem::Run()
+{
+	auto view = Engine::Registry().view<Transform, SimpleCollision, Collectible>();
+
+	for (auto entity : view)
+	{
+		auto& col = view.get<SimpleCollision>(entity);
+		auto& collectible = view.get<Collectible>(entity);
+
+		if (col.colided)
+		{
+			if (Engine::Registry().has<TankCharacter>(col.colidedWith))
+			{
+				auto& tank = Engine::Registry().get<TankCharacter>(col.colidedWith);
+				if (collectible.collType == ECollectibleType::Coin)
+				{
+					tank.coins++;
+					m_NumCoins--;
+				}
+				else
+				{
+					if (collectible.collType == ECollectibleType::ExtraLife)
+					{
+						tank.lives++;
+					}
+					if (collectible.collType == ECollectibleType::FireRateUp)
+					{
+						tank.fireRate = 2;
+						tank.powerDuration += 4;
```

Comment:
good thing to do here is to make this values into const variables with proper names.

Reply:
Ok, thanks! I'll do this.